src/main/java/com/bnkrll/controller/SessionController.java

```diff
@@ -35,4 +35,18 @@ public List<Session> getPreviousSessions(
                     int last) {
         return sessionRepository.getLastSessions(last);
     }
+
+    @PatchMapping
+    public ResponseEntity<Session> updateSession(@RequestBody Session session){
+        List<Session> lastSession = sessionRepository.getLastSessions(1);
```

Comment:
Will there ever be a time when the user will need to update a previous session after already starting a new session?

If so, then maybe we should have the UI pass in the session Id and find the session with that id rather than just getting the last session.

Reply:
No, there will never be a time for such scenario. 

The method name was misleading, so I changed it from `getLastSessions(1)` to `getSessions(1)`

Reply:
I think the method name getSessions is worse. Maybe getPreviousSessions(1). Alternatively you could add a specific query in the repository to get only the last session.

Reply:
Ok , getPreviousSessions(1) is the name of the GET controller, don't think that will work 

```
    @GetMapping
    public List<Session> getPreviousSessions(
            @RequestParam(value = "last", defaultValue = "30")
                    int last) {
        return sessionRepository.getSessions(last);
    }
```

Reply:
getCurrentSesssion maybe ? I'll add that specific query for the update usecase

Reply:
They can be the same name. I think the original name was fine. You just need a getLastSession

Reply:
Decided to go with the method below, since its already in the current repository implementation

```
    @Override
    public Session findById(String sessionID) {
        return this.data.get(sessionID);
    } 
```